python/tests/test_adjoint_solver.py

```diff
@@ -943,6 +943,38 @@ def test_mode_source_bandwidth(self):
                 self.assertLessEqual(rel_err, tol)
                 print(f"PASSED: fwidth={fwidth:.5f}, m={m}, err={rel_err:.10f}")
 
+    def test_reverse_design(self):
+        """Verifies that the reverse_design finds better initialization for specific
+        design than that design itself as initialization."""
+        print("*** TESTING reverse_design ***")
```

Comment:
This is too loose of a test.

I would do a [gold master test](https://codeclimate.com/blog/gold-master-testing) where you just verify that you reproduce the results of a previous run that you checked manually.